tracer/src/Datadog.Trace/Sampling/AgentSamplingRule.cs

```diff
@@ -38,40 +37,46 @@ internal class AgentSamplingRule : ISamplingRule
 
         public float GetSamplingRate(Span span)
         {
-            Log.Debug("Using the default sampling logic");
-            float defaultRate;
+            var service = span.ServiceName;
+            var env = span.Context.TraceContext.Environment ?? string.Empty;
 
-            if (_sampleRates.Count == 0)
+            if (_sampleRates.Count > 0)
             {
-                // either we don't have sampling rate from the agent yet (cold start),
-                // or the only rate we received is for "service:,env:", which is not added to _sampleRates
-                defaultRate = _defaultSamplingRate ?? 1;
+                var key = new SampleRateKey(service, env);
 
-                // add the _dd.agent_psr tag even if we didn't use rates from the agent (to ease investigations)
-                SetSamplingRateTag(span, defaultRate);
-                return defaultRate;
+                if (_sampleRates.TryGetValue(key, out var matchingRate))
+                {
+                    // found a matching sampling rate by service/env
+                    Log.Debug(
```

Comment:
Do we _need_ these  debug logs? I just ask because I've come to realize we generate 100s of MBs of logs in Debug mode if there's any load on a site, so we should probably start being more judicious 😬

I guess we're not writing any _more_ logs, just something more useful, so meh

Reply:
Yeah, I'm happy to remove logs if that's your recommendation. Should we aim for a _single_ sampling log entry per trace? Or is that too much?

Reply:
Ok, I remove most of the sampling debug logs.